src/App.tsx

```diff
@@ -1,26 +1,266 @@
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
+        <Board values={vals} />
       </header>
     </div>
   );
+};
+
+type SquareAddress = number;
+
+interface BoardState {
+    highlights: SquareAddress[];
+    contradicts: SquareAddress[];
+    permanentValues: SquareAddress[];
+    values: Array<SquareValue | null>;
+}
+
+interface BoardProps {
+    values: Array<SquareValue | null>;
+}
+
+class Board extends React.Component<BoardProps, BoardState> {
+    public state: BoardState = {
+        highlights: [37],
+        contradicts: [38],
+        permanentValues: [],
+        values: this.props.values,
+        /**
+          * //TODO: CREATE marking state and transfer marking logic to board state.
+          */
+    };
+
+    /**
+      * // TODO: other highlight functionality
+      */
+    public toggleSelectSquare(i: SquareAddress) {
+        this.setState( {
+            ...this.state,
+            highlights: [i],
+            },
+        );
+    }
+
+    public getPermanentValues(): SquareAddress[] {
+        const list = [];
+        for (let i = 0; i < 81; i++) {
+            if (this.props.values[i]) {
+                list.push(i as SquareAddress);
+            }
+        }
+        return list;
+    }
+
+    public normalMarkSelectedSquares(i: SquareValue) {
+        for (let j = 0; j < this.state.highlights.length; j++) {
+            const addy: SquareAddress = this.state.highlights[j];
+            if (!this.state.permanentValues.includes(addy)) {
+                const newValues = JSON.parse(JSON.stringify(this.state.values));
+                newValues[addy] = i;
+                this.setState({
+                    values: newValues,
+                });
+            }
+        }
+    }
+
+    public deleteSelectedSquares(): void {
+        const newValues = JSON.parse(JSON.stringify(this.state.values));
+        for (const address of this.state.highlights) {
+            if (!this.state.permanentValues.includes(address)) {
+                newValues[address] = null;
+                /**
+                  * //TODO: if marking exist on first delete click
+                  * delete only value and show markings on second
+                  * click delete markings
+                  */
+            }
+        }
+        this.setState({
+            values: newValues,
+        });
+    }
+
+/*     convertUndefinedValues(): (SquareValue | null)[] {
+        let list = [];
+        for (var i = 0; i < 81; i++) {
+            if (this.props.values[i] == undefined) {
+                list.push(null);
+            } else {
+                list.push(this.props.values[i] as SquareValue);
+            }
+        }
+        return list;
+    } */
+
+    public render() {
+        this.state.permanentValues = this.getPermanentValues();
```

Comment:
Rule: Don't ever set state via `this.state.foo = bar`. Always call `setState`.

Rule: Don't ever call `setState` in `render`